CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+  - one `severity:<bug_severity>` if possible, if severity is not known leave it blank it will be added later by another team member
+- Contributors can change issues' tags (severity, area, kind, ...) if necessary. 
+  Keep in mind that explaining why you are editing an issue is the best way to avoid misunderstanding.
+- If you are working on a bug or plan to work on a bug, assign yourself to the bug.
+  If a bug is assigned, consider that they are already working on the resolution.
+- PRs solving bugs should add a regression tests to ensure that bug will not be back in the future.
+
+### General code recommendations
+
+- Take the time to pick good names.
+  Avoid non well-known abbreviations.
+
+### Python
+
+- Follow [the Zen of Python](https://www.python.org/dev/peps/pep-0020/).
+- To keep our API surface minimal, only make public what is necessary.
+- Documentation should only be there to explain things that aren't self evident.
+- Code is linted with [flake8](https://github.com/csachs/pyproject-flake8).
+- Code is formatted with [Black](https://github.com/psf/black).
+- Imports are sorted with [Isort](https://github.com/PyCQA/isort).
+
+### Test
+
+- Python tests are written using [pytest](https://docs.pytest.org/).
+
+### Rust
+
+- To keep our API surface minimal, only make public what is necessary.
+- Documentation should only be there to explain things that aren't self evident.
+- Use the [documentation example](https://doc.rust-lang.org/rust-by-example/meta/doc.html) to know how to phrase and format your documentation.
+- Code is linted with [clippy](https://github.com/rust-lang/rust-clippy).
+- Code is formatted with [fmt](https://github.com/rust-lang/rustfmt).
+
+### Test
+
+- Test as much code as possible to avoid future regression and show that added code is bullet proof.
+- Rust code is tested files per files following these [recommendations](https://doc.rust-lang.org/book/ch11-01-writing-tests.html).
+
+### Java
+
+- To keep our API surface minimal, only make public what is necessary.
+- All public methods must be documented.
+  Avoid paraphrasing doc.
+- Code is formatted with [checkstyle](https://checkstyle.sourceforge.io/).
+
+#### Tests
+
+- TODO
+
+### Javascript / Typescript / Front
+
+- To keep our API surface minimal, only make public what is necessary.
+- Documentation should only be there to explain things that aren't self evident.
+- When adding new files, write them in TypeScript as there is a goal to move to TypeScript.
+- Code is linted with [eslint](https://eslint.org/).
+- Code is formatted with [prettier](https://prettier.io/).
+
+#### Tests
+
+- End-to-end tests are required for stable and critical features.
+  [Playwright](https://playwright.dev/) is used to write these tests.
+- To write unit test use [vitest](https://vitest.dev/).
+
+### Integration tests
+
+- Integration tests are written using [pytest](https://docs.pytest.org/) under the `/tests` folder.
+- Test as much as you can when adding a new api.
```

Comment:
Move to the "general code recommendations" section ? Or maybe move the "test" section under "general code recommendations" ?

Reply:
I think this part will be improved with future integration tests improvements like https://github.com/DGEXSolutions/osrd/issues/3465